v23.1/cdc-queries.md

```diff
@@ -253,62 +325,83 @@ For the previous JSON example:
-'accounts' AS table,
-IF (cdc_is_delete(),'delete','create') AS type,
-jsonb_build_object('owner', owner) AS data
-FROM accounts;
+  event_schema_timestamp()::int AS event_timestamp,
+  'accounts' AS table,
+  event_op() AS type,
+  jsonb_build_object('owner', owner) AS data
+  FROM accounts;
 ~~~
 
-For a different usage of the outbox pattern, you may still want an events table to track and manage the lifecycle of an event. You can also use CDC transformations in this case to filter the event management metadata out of a message. 
+For a different usage of the outbox pattern, you may still want an events table to track and manage the lifecycle of an event. You can also use CDC queries in this case to filter the event management metadata out of a message. 
 
 For example, when you delete a message in your outbox table after processing it (or with [row-level TTL](row-level-ttl.html)). You can filter the [delete messages](#filter-delete-messages) from your changefeed:
 
 {% include_cached copy-clipboard.html %}
 ~~~sql
-CREATE CHANGEFEED INTO ‘kafka://endpoint?topic_name=events’ AS SELECT * FROM outbox WHERE NOT cdc_is_delete();
+CREATE CHANGEFEED INTO 'kafka://endpoint?topic_name=events' AS SELECT * FROM outbox WHERE event_op() != 'delete';
 ~~~
 
 Similarly, if you have a status column in your outbox table tracking its lifecycle, you can filter out updates as well so that only the initial insert sends a message:
 
 {% include_cached copy-clipboard.html %}
 ~~~sql
-CREATE CHANGEFEED INTO 'scheme://sink-URI' WITH schema_change_policy='stop' AS SELECT status FROM outbox WHERE cdc_prev()->'status' IS NULL;
+CREATE CHANGEFEED INTO 'scheme://sink-URI' AS SELECT status, cdc_prev FROM outbox WHERE (cdc_prev).status IS NULL;
+~~~
+
+Since all non-primary key columns will be `NULL` in the `cdc_prev` output for an insert message, insert messages will be sent. Updates will not send, as long as the status was not previously `NULL`.
+
+### Queries and user-defined functions
+
+{% include_cached new-in.html version="v23.1" %} You can create CDC queries that include [user-defined functions](user-defined-functions.html).
+
+The following [`CREATE FUNCTION`](create-function.html) statement builds the `doubleRevenue()` function at the database level: 
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE FUNCTION doubleRevenue(r int)
+  RETURNS INT IMMUTABLE LEAKPROOF LANGUAGE SQL AS 
+  $$ SELECT 2 * r $$;
 ~~~
 
-Since all non-primary key columns will be `NULL` in the [`cdc_prev()`](#cdc-transformation-function-support) output for an insert message, insert messages will be sent. Updates will not send, as long as the status was not previously `NULL`.
+You can then use this function within a CDC query tagetting a table in the same database:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE CHANGEFEED INTO 'external://sink' AS SELECT rider_id, doubleRevenue(rides.revenue::int) FROM rides WHERE revenue < 30;
```

Comment:
Haha, if only it were this easy in real life! ( i.e. `doubleRevenue(thing.revenue::int)` )

Great example!